controllers/backstage_service.go

```diff
@@ -33,14 +33,19 @@ func (r *BackstageReconciler) reconcileBackstageService(ctx context.Context, bac
 			Namespace: ns,
 		},
 	}
+
 	if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, service, r.serviceObjectMutFun(ctx, service, *backstage,
 		backstage.Spec.RawRuntimeConfig.BackstageConfigName, "service.yaml", service.Name, service.Name)); err != nil {
+
 		if errors.IsConflict(err) {
 			return retryReconciliation(err)
 		}
 		msg := fmt.Sprintf("failed to deploy Backstage Service: %s", err)
+
 		setStatusCondition(backstage, bs.ConditionDeployed, metav1.ConditionFalse, bs.DeployFailed, msg)
+		return fmt.Errorf("%s %w", msg, err)
```

Comment:
What about adding a test case for it?

Reply:
Good idea, please add note to the issue, will do in a separate PR later.

Reply:
Sounds good. I added a task to issue: https://github.com/janus-idp/operator/issues/202